### PR TITLE
Move KUBE_GIT_UPSTREAM out of init.sh and into *-munge-docs.sh.

### PR DESCRIPTION
**Pull request by david-mcmahon**
It is only used in those 2 scripts and this way we can set the value dynamically.
Clean up a bit too (80col, formatting)

### CODE REVIEW

**Comment by david-mcmahon** [comment]
Well that's not going to work then.

File: hack/update-munge-docs.sh [hunk 1]
```diff
@@ -21,6 +21,9 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
+git_upstream=$(kube::util::git_upstream_remote_name)
```
Comment (lavalamp) [inline hack/update-munge-docs.sh]: I suggest continuing to use the env variable, but defaulting it to this value if it's not set.

Comment (david-mcmahon) [inline hack/update-munge-docs.sh]: If we can get rid of another unused global, I think we're making the world a better place.  If you grep through hack/*, KUBE_GIT_UPSTREAM is set once in init and referred to only in the -munge- scripts and documented exactly nowhere.  My inclination here is to simplify and reduce.  But if you feel strongly, I can put it back.

**Comment by david-mcmahon** [comment]
ping.

File: hack/update-munge-docs.sh (hunk 1 above)
Comment (lavalamp) [inline hack/update-munge-docs.sh]: ah yeah, the part I was objecting to wasn't clobbering the env var so much as it's no longer possible to set the option to a custom value at all. If you don't think that's ever a reasonable use case then I'm fine with this.

Comment (david-mcmahon) [inline hack/update-munge-docs.sh]: I feel like we're really overloaded with globals in the scripts and it's not clear to me how or if they are used at all.  Given it isn't documented anywhere I'm inclined to beg forgiveness here and scrub it.  We can always add it back.

**Comment by lavalamp** [comment]
LGTM-- if people show up with pitchforks, I will direct them to your desk :)

**Comment by david-mcmahon** [comment]
Please do! I will invite them to return the global with documentation in a new PR.

**Pull request merged**